File: app/src/main/java/org/odddev/fantlab/core/di/AppModule.java
package org.odddev.fantlab.core.di;

import android.content.Context;
import android.support.annotation.NonNull;

import org.odddev.fantlab.core.network.INetworkChecker;
import org.odddev.fantlab.core.network.NetworkChecker;
import org.odddev.fantlab.core.rx.ISchedulersResolver;
import org.odddev.fantlab.core.rx.SchedulersResolver;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author kenrube
 * @since 23.08.16
 */

@Module
public class AppModule {

    private final Context mContext;

    public AppModule(Context context) {
        mContext = context;
    }

    @Singleton
    @Provides
    @NonNull
    Context getContext() {
        return mContext;
    }

    @Singleton
    @Provides
    ISchedulersResolver provideSchedulersResolver() {
        return new SchedulersResolver();
    }

    @Singleton
    @Provides
    INetworkChecker provideNetworkChecker() {
        return new NetworkChecker();
    }
}

File: app/src/main/kotlin/org/odddev/fantlab/core/di/AppModule.kt
package org.odddev.fantlab.core.di

import android.arch.persistence.room.Room
import android.content.Context
import dagger.Module
import dagger.Provides
import org.odddev.fantlab.core.db.MainDatabase
import org.odddev.fantlab.core.network.INetworkChecker
import org.odddev.fantlab.core.network.NetworkChecker
import javax.inject.Singleton

@Module
class AppModule(internal val context: Context) {

	@Singleton
	@Provides
	fun getContext(): Context = context

	@Singleton
	@Provides
	fun provideNetworkChecker(): INetworkChecker = NetworkChecker()

	@Singleton
	@Provides
	fun provideMainDatabase(): MainDatabase = Room
			.databaseBuilder(context, MainDatabase::class.java, "main.db")
			.build()
}
